### PR TITLE
[DOCS] Rewrite analysis intro

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes
* Rewrites 'Text analysis' page intro as high-level definition.
  Adds guidance on when users should configure text analysis
* Rewrites and splits index/search analysis content:
  * Conceptual content -> 'Index and search analysis' under 'Concepts'
  * Task-based content -> 'Specify an analyzer' under 'Configure...'
* Adds detailed examples for when to use the same index/search analyzer
  and when not.
* Adds new example snippets for specifying search analyzers

### Previews
* **Text analysis intro:**
   http://elasticsearch_51184.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/analysis.html
* **Index and search analysis (concepts):**
   http://elasticsearch_51184.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/analysis-index-search-time.html
* **Specify an analyzer (task):**
   http://elasticsearch_51184.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/specify-analyzer.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@debadair Any feedback on this one? Thanks!

File: docs/reference/analysis.asciidoc [hunk 1]
```diff
@@ -4,141 +4,42 @@
...
+optimized for search.
 
 [float]
-== Index time analysis
+[[when-to-configure-analysis]]
+=== When to configure text analysis
 
-For instance, at index time the built-in <<english-analyzer,`english`>> _analyzer_ 
-will first convert the sentence:
+{es} comes with smart defaults for text analysis. These defaults work well for
```
Comment (mayya-sharipova) [inline docs/reference/analysis.asciidoc]: smart defaults?   ES just uses a default analyzer for text fields,  which many not be the best option for many use cases.

File: docs/reference/analysis.asciidoc [hunk 2]
```diff
@@ -4,141 +4,42 @@
...
-"The QUICK brown foxes jumped over the lazy dog!"
-------
+If you're happy with your search experience, no further setup is needed;
+you can skip the chapters in this section.
 
-into distinct tokens. It will then lowercase each token, remove frequent
-stopwords ("the") and reduce the terms to their word stems (foxes -> fox,
-jumped -> jump, lazy -> lazi). In the end, the following terms will be added
-to the inverted index:
+However, there are less common cases where configuring text analysis is
```
Comment (mayya-sharipova) [inline docs/reference/analysis.asciidoc]: I don't think that configuring an analyzer is an uncommon use case,  I would think it should be quite common for text fields.

File: docs/reference/analysis/specify-analyzer.asciidoc [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+[TIP]
+.Keep it simple
+====
+The flexibility to specify analyzers at different levels and for different times
+is great... _but only when it's needed_.
+
+In most cases, a simple approach works best: Specify an analyzer for each
+`text` field, as outlined in <<specify-index-field-analyzer>>. No other
+analyzers need to be specified.
```
Comment (mayya-sharipova) [inline docs/reference/analysis/specify-analyzer.asciidoc]: >  No other analyzers need to be specified.

This is not very clear for me.  Sorry, if I misinterpreted this paragraph.

 Analyzers are only specified for text fields, so it is impossible to specify analyzers for any other field types.

File: docs/reference/analysis/index-search-time.asciidoc [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+|===
+|Token      | `appli`      | `apple`
+|`a`        | X            | X
+|`ap`       | X            | X
+|`app`      | X            | X
+|`appl`     | X            | X
+|`appli`    |              | X
+|===
+
+This means the search would erroneously match `apple`. Not only that, it would
```
Comment (mayya-sharipova) [inline docs/reference/analysis/index-search-time.asciidoc]: This is a very good example.

Another example could be a use case with synonyms, where we specify synonym filter only during search, as it is redundant to use synonyms both at indexing and querying.

**Review by mayya-sharipova — APPROVED** [review]
@jrodewig Thanks, great PR.
I can't comment and review the organization of files, but the content LGTM.

File: docs/reference/analysis.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/analysis.asciidoc]: Good point! It's clearer if we just directly state that that if you use `text` fields, take a look. If not, go ahead and skip this section.

Made those changes with a2f08d21e0a93a245e19254118e4ece83d06d080.

File: docs/reference/analysis/specify-analyzer.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/analysis/specify-analyzer.asciidoc]: That sentence was referring to specifying an index analyzer or field-level search analyzer for a field. However, I agree with you: this paragraph is clearer without that sentence. Thanks!

**Comment by jrodewig** [comment]
Thanks so much for your review @mayya-sharipova!

**Comment by jrodewig** [comment]
### Backport commits
master 3c28a10 
7.x 4fcf5a9de4a
7.6 2e0ed041169
7.5 a6edbbdf94d

**Pull request merged**